### PR TITLE
fix: some return types

### LINKED ISSUE #305: Vet v6.0.0-alpha.5 types emitted types against v5.5.3
We need to ensure that the types emitted from our packages are API compatible with the previous DefinitelyTyped package.

See:

1. #166 
2. #198 
3. `faker.address.nearbyGPSCoordinate` (`coordinate` is a ReadOnlyArray in DefinitelyTyped)
4. `faker.address.companyName` has an arg of `format` which should be `string | number`
5. `faker.database.column` is implicitly return typed as `string` in 6.0.0-alpha.5 but should be set to return `string`
7. `faker.database.type()` is implicitly return typed as `string` in 6.0.0-alpha.5 but should be set to return `string`
8. `faker.database.collation()` is implicitly return typed as `string` in 6.0.0-alpha.5 but should be set to return `string`
9. `faker.database.engine()` is implicitly return typed as `string` in 6.0.0-alpha.5 but should be set to return `string`
10. `faker.finance.currencyCode()` is return typed `any`
11. `faker.finance.currencyName()` is return typed `any`
12. `faker.finance.currencySymbol()` is return typed `any`
13. `faker.hacker.abbreviation()` is implicitly return typed as `string`
14. `faker.hacker.adjective()` is implicitly return typed as `string`
15. `faker.hacker.noun()` is implicitly return typed as `string`
16. `faker.hacker.verb()` is implicitly return typed as `string`
17. `faker.hacker.ingverb()` is implicitly return typed as `string`
18. `faker.hacker.phrase()` is implicitly return typed as `string`

I'm sure there are a few others, but we need to check type by type to ensure spec compatibility.

### PR DESCRIPTION
**Pull request by ejcheng**
Partially fixes #305

### CODE REVIEW

**Comment by Shinigami92** [comment]
Currently one of our problems is that `arrayElement` doesn't guarantee type-safetynes (regrading to compile-time).
Our `definitions` currently are just _mostly_ arrays, but could be anything :shrug: 
I know that there are currently mostly `string[]` and we could just take over the types of `@types/faker`, but I would like to double-check that instead of assuming.

**Review by ST-DDT — DISMISSED** [review]

**Comment by ST-DDT** [comment]
IMO we should set these to the expected return types. Double checking with the existing ones would still be a good idea.

**Review by doughlass — DISMISSED** [review]

**Comment by import-brain** [comment]
> IMO we should set these to the expected return types. Double checking with the existing ones would still be a good idea.

All of the ones in this PR were checked against @types/faker latest version, they all seem reasonable and match up.

**Comment by ST-DDT** [comment]
A probably wrong commit ended up here.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 blank_issues_enabled: false
 contact_links:
   - name: Discord Chat
-    url: discord.gg/4qDjAmDj4P
+    url: chat.fakerjs.dev
```
Comment (ST-DDT) [inline .github/ISSUE_TEMPLATE/config.yml:4]: Please remove the last commit here.

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (import-brain) [reply .github/ISSUE_TEMPLATE/config.yml]: Accidentally committed the chat.fakerjs.dev changes to this branch, just reset and force-pushed

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Comment by demipel8** [comment]
Wrong mention, sorry.

**Pull request merged**